src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -488,12 +488,47 @@ public bool SpansPreprocessorDirective(IEnumerable<SyntaxToken> tokens)
         private bool SpansPreprocessorDirective(SyntaxTriviaList list)
             => list.Any(t => IsPreprocessorDirective(t));
 
-        public bool IsInHeader(SyntaxNode node, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader)
+        public bool IsOnHeader(int position, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader)
         {
             var start = GetStartOfNodeExcludingAttributes(ownerOfHeader);
             var end = lastTokenOrNodeOfHeader.FullSpan.End;
 
-            return node.Span.Start >= start && node.Span.End <= end;
+            return start <= position && position <= end;
+        }
+
+        public bool IsInHeader(int position, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader, ImmutableArray<SyntaxNode> holes)
+        {
+            var inHeader = IsOnHeader(position, ownerOfHeader, lastTokenOrNodeOfHeader);
+            if (!inHeader)
+            {
+                return false;
+            }
+
+            // Holes are exclusive: to return false it needs to be _inside_ a hole not only on the edge
+            // -> to be consistent with other 'being on the edge' of Tokens/Nodes.
+            if (holes.Any(h => (h.Span.Start < position && position < h.Span.End)))
```

Comment:
nit: `return holes.All(...)`

Reply:
fwiw, i find the extra parens here *more* confusing.  because i expected tehy were needed for precedence, then couldn't find the close paren in the expression itself.

Reply:
I see the thing about parantheses but TBH not having it there makes it hard for me to see what the expression is.

Reply:
As for the first point, after rewriting it just doesn't look as obvious for some reason. The explicit, if I'm in any hole -> return false else return true is more readable then return negation if am I in any hole.